mailpoet/tests/performance/tests/lists-complex-segment.js

```diff
@@ -27,100 +32,108 @@ export async function listsComplexSegment() {
   });
   const page = browser.newPage();
 
-  try {
-    const complexSegmentName =
-      'Complex Segment ' + Math.floor(Math.random() * 9999 + 1);
+  const complexSegmentName =
+    'Complex Segment ' + Math.floor(Math.random() * 9999 + 1);
 
-    // Go to the page
-    await page.goto(
-      `${baseURL}/wp-admin/admin.php?page=mailpoet-segments#/lists`,
-      {
-        waitUntil: 'networkidle',
-      },
-    );
+  // Go to the page
+  await page.goto(
+    `${baseURL}/wp-admin/admin.php?page=mailpoet-segments#/lists`,
+    {
+      waitUntil: 'networkidle',
+    },
+  );
 
-    // Log in to WP Admin
-    authenticate(page);
+  // Log in to WP Admin
+  authenticate(page);
 
-    // Wait for async actions
-    await page.waitForNavigation({ waitUntil: 'networkidle' });
+  // Wait for async actions
+  await page.waitForNavigation({ waitUntil: 'networkidle' });
 
-    // Click to add a new segment
-    page.waitForSelector('[data-automation-id="new-segment"]');
-    page.locator('[data-automation-id="new-segment"]').click();
-    page
-      .locator('[data-automation-id="input-name"]')
-      .type(complexSegmentName, { delay: 25 });
+  // Click to add a new segment
+  page.waitForSelector('[data-automation-id="new-segment"]');
+  page.locator('[data-automation-id="new-segment"]').click();
+  page
+    .locator('[data-automation-id="input-name"]')
+    .type(complexSegmentName, { delay: 25 });
 
-    // Select "Subscribed to a list" action
-    selectInReact(page, '#react-select-2-input', 'subscribed to list');
-    selectInReact(page, '#react-select-4-input', defaultListName);
-    page.waitForSelector('.mailpoet-form-notice-message');
-    page.waitForLoadState('networkidle');
+  // Select "Subscribed to a list" action
+  selectInReact(page, '#react-select-2-input', 'subscribed to list');
+  selectInReact(page, '#react-select-4-input', defaultListName);
+  page.waitForSelector('.mailpoet-form-notice-message');
+  describe(listsPageTitle, () => {
+    describe('should be able to see calculating message', () => {
+      expect(
+        page.locator('.mailpoet-form-notice-message').innerText(),
+      ).to.contain('Calculating segment size…');
+    });
+  });
+  page.waitForLoadState('networkidle');
 
-    // Click to add a new segment action
-    page
-      .locator(
-        '#segments_container > form > div > div.mailpoet-segments-segments-section > button',
-      )
-      .click();
+  // Click to add a new segment action
+  page
+    .locator(
+      '#segments_container > form > div > div.mailpoet-segments-segments-section > button',
+    )
+    .click();
 
-    // Select "Subscribed date" action
-    selectInReact(page, '#react-select-5-input', 'subscribed date');
+  // Select "Subscribed date" action
+  selectInReact(page, '#react-select-5-input', 'subscribed date');
+  page.waitForSelector('.mailpoet-form-notice-message');
+  describe(listsPageTitle, () => {
```

Comment:
I wonder if checking for "Calculating segment size" makes the test `flaky` as it is time constrained. I mean, it appears and disappears.

Imo it shouldn't be flaky at this point.

Reply:
I understand your worry. Haven't seen it as flaky in multiple test executions, however, if I see it one time as flaky I will remove it.